core/server/apps/amp/lib/router.js

```diff
@@ -63,8 +63,8 @@ function getPostData(req, res, next) {
         }));
     }
 
-    helpers.postLookup(urlWithoutSubdirectoryWithoutAmp, {permalinks: permalinks})
-        .then(function handleResult(result) {
+    helpers.postLookup(urlWithoutSubdirectoryWithoutAmp, {permalinks})
+        .then((result) => {
```

Comment:
In my opinion, I don't think we should use arrow functions for promise chains in general - only if you need to have to use the correct `this` context. Otherwise the error stack won't contain any useful debug information.

```
callSomething()
    .then(() => {
        throw new Error('error appeared');
    })
    .catch((err) => {
        console.log(err);
    });
```

```
Error: error appeared
    at request.then (/Users/kirrg001/dev/ghost/ghost-workspace2/test.js:6:8)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
```

VS.

```
callSomething()
    .then(function updateUser() {
        throw new Error('error appeared');
    })
    .catch(function onError(err) {
        console.log(err);
    });
```

```
Error: error appeared
    at updateuser (/Users/kirrg001/dev/ghost/ghost-workspace2/test.js:6:8)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
```

You can read the line of the file, but it's much easier to read the function name in my opinion?

cc @kevinansfield

Reply:
@kirrg001 @kevinansfield That's why I have asked a question about named function or not in the original issue (#9667) but no one has answered yet. It's more about coding convention than logic. If you always want to know function name in the call stack so arrow function has no chance anywhere. And if we don't make it clear, others will convert callbacks to arrows too because it's ES6 :)

Reply:
In Ghost-Admin we never use named functions, always preferring arrow syntax but that may be more to do with the way that `this` is used there and that the browser's built-in debugging means named functions aren't so useful.

@kirrg001 I think this one is up to you and what makes more sense in the Ghost context 🙂